src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IEventAssignmentExpression.cs

```diff
@@ -0,0 +1,250 @@
+      Instance Receiver: null
+  Handler: IMethodBindingExpression: void C.Handler(System.Object sender, System.EventArgs e) (OperationKind.MethodBindingExpression, Type: System.EventHandler) (Syntax: 'Handler')
+      Instance Receiver: IInstanceReferenceExpression (InstanceReferenceKind.Implicit) (OperationKind.InstanceReferenceExpression, Type: C) (Syntax: 'Handler')
+";
+            var expectedDiagnostics = new[] { 
+                // file.cs(6,38): warning CS0067: The event 'Test.MyEvent' is never used
+                //     public static event EventHandler MyEvent;
+                Diagnostic(ErrorCode.WRN_UnreferencedEvent, "MyEvent").WithArguments("Test.MyEvent").WithLocation(6, 38)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<AssignmentExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact]
+        public void AddEventHandler_DelegateTypeMismatch()
+        {
+            string source = @"
+using System;
+
+class Test
+{
+    public event EventHandler MyEvent;    
+}
+
+class C
+{
+    void Handler(object sender)
+    {
+    } 
+
+    void M()
+    {
+        var t = new Test();
+        /*<bind>*/t.MyEvent += Handler/*<bind>*/;
+    }
+}
+";
+            string expectedOperationTree = @"
+IEventAssignmentExpression (EventAdd)) (OperationKind.EventAssignmentExpression, Type: System.Void, IsInvalid) (Syntax: 't.MyEvent += Handler')
+  Event Reference: IEventReferenceExpression: event System.EventHandler Test.MyEvent (OperationKind.EventReferenceExpression, Type: System.EventHandler, IsInvalid) (Syntax: 't.MyEvent')
+      Instance Receiver: ILocalReferenceExpression: t (OperationKind.LocalReferenceExpression, Type: Test, IsInvalid) (Syntax: 't')
+  Handler: IConversionExpression (ConversionKind.Invalid, Implicit) (OperationKind.ConversionExpression, Type: System.EventHandler, IsInvalid) (Syntax: 'Handler')
+      Operand: IOperation:  (OperationKind.None, IsInvalid) (Syntax: 'Handler')
+";
+            var expectedDiagnostics = new[] {                  
+                // file.cs(18,19): error CS0123: No overload for 'Handler' matches delegate 'EventHandler'
+                //         /*<bind>*/t.MyEvent += Handler/*<bind>*/;
+                Diagnostic(ErrorCode.ERR_MethDelegateMismatch, "t.MyEvent += Handler").WithArguments("Handler", "System.EventHandler").WithLocation(18, 19),
+                // file.cs(6,31): warning CS0067: The event 'Test.MyEvent' is never used
+                //     public event EventHandler MyEvent;
+                Diagnostic(ErrorCode.WRN_UnreferencedEvent, "MyEvent").WithArguments("Test.MyEvent").WithLocation(6, 31)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<AssignmentExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact]
+        public void AddEventHandler_AssignToStaticEventOnInstance()
```

Comment:
For completeness, please test a reference to a instance event without a receiver, and test the same two cases in VB.